Tasks/AzureRmWebAppDeployment/AzureRmWebAppDeployment.ps1

```diff
@@ -60,20 +62,29 @@ try{
     # Ensure that at most a package (.zip) file is found
     $packageFilePath = Get-SingleFilePath -file $Package
 
-    if($XmlTransformation -eq "true")
+    if( $XmlTransformation -eq "true" -or $VariableSubstitution -eq "true")
     {
         # Unzip the source package
         $unzippedPath = UnzipWebDeployPkg -PackagePath $packageFilePath
-        # Search for all the web.config files
-        $webconfigFiles = Find-VstsFiles -LegacyPattern "$unzippedPath\**\web.config" -IncludeFiles
-        # Foreach web.config file apply Web.Release.Config and Web.Environment.config
-        foreach ($configFile in $webconfigFiles) {
-            FindAndApplyTransformation -baseFile $configFile -tranformFile "web.release.config" -xdtFilesRoot $XdtFilesRoot
-            if($env:RELEASE_ENVIRONMENTNAME.config)
-            {
-                FindAndApplyTransformation -baseFile $configFile -tranformFile "web.$env:RELEASE_ENVIRONMENTNAME.config" -xdtFilesRoot $XdtFilesRoot
+        
+        if( $XmlTransformation -eq "true" )
+        {
+            # Search for all the web.config files
+            $webconfigFiles = Find-VstsFiles -LegacyPattern "$unzippedPath\**\web.config" -IncludeFiles
+            # Foreach web.config file apply Web.Release.Config and Web.Environment.config
+            foreach ($configFile in $webconfigFiles) {
+                FindAndApplyTransformation -baseFile $configFile -tranformFile "web.release.config" -xdtFilesRoot $XdtFilesRoot
+                if($env:RELEASE_ENVIRONMENTNAME.config)
+                {
+                    FindAndApplyTransformation -baseFile $configFile -tranformFile "web.$env:RELEASE_ENVIRONMENTNAME.config" -xdtFilesRoot $XdtFilesRoot
+                }
             }
-        }
+         }
+         if( $VariableSubstitution -eq "true" )
+         {
+            Substitute-Variables -WebAppFolderPath $unzippedPath -ConfigFileRegex "web.*config"
```

Comment:
The spec says we should support any *.config. We are passing web.*config. Are we on same page?